sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteFilesTest.java

```diff
@@ -449,16 +455,23 @@ public String apply(String input) {
   @Test
   @Category(NeedsRunner.class)
   public void testDynamicDestinationsBounded() throws Exception {
-    testDynamicDestinationsHelper(true);
+    testDynamicDestinationsHelper(true, false);
   }
 
   @Test
   @Category(NeedsRunner.class)
   public void testDynamicDestinationsUnbounded() throws Exception {
-    testDynamicDestinationsHelper(false);
+    testDynamicDestinationsHelper(false, false);
   }
 
-  private void testDynamicDestinationsHelper(boolean bounded) throws IOException {
+  @Test
+  @Category(NeedsRunner.class)
+  public void testDynamicDestinationsFillEmptyShards() throws Exception {
+    testDynamicDestinationsHelper(true, true);
```

Comment:
no, because we don't fill empty shards on windowed writes (long discussion - but basically matching the batch semantics would mean noticing "missing windows" and that is not a well-defined concept outside of a very specific set of window functions).